site/app/libraries/database/PostgresqlDatabaseQueries.php

```diff
@@ -67,6 +67,49 @@ public function getUserById($user_id) {
         return new User($this->core, $details);
     }
 
+    public function getUserByNumericId($numeric_id) {
+        $this->submitty_db->query("SELECT * FROM users WHERE user_numeric_id=?", array($numeric_id));
+
+        if ($this->submitty_db->getRowCount() === 0) {
+            return null;
+        }
+
+        $details = $this->submitty_db->row();
+
+        if ($this->course_db) {
+            $this->course_db->query("
+SELECT u.*, ns.merge_threads, ns.all_new_threads,
+       ns.all_new_posts, ns.all_modifications_forum,
+       ns.reply_in_post_thread,ns.team_invite,
+       ns.team_member_submission, ns.team_joined,
+       ns.self_notification,
+       ns.merge_threads_email, ns.all_new_threads_email,
+       ns.all_new_posts_email, ns.all_modifications_forum_email,
+       ns.reply_in_post_thread_email, ns.team_invite_email, 
+       ns.team_member_submission_email, ns.team_joined_email,
+       ns.self_notification_email,sr.grading_registration_sections
+       
+FROM users u
+LEFT JOIN notification_settings as ns ON u.user_numeric_id = ns.user_numeric_id
+LEFT JOIN (
+  SELECT array_agg(sections_registration_id) as grading_registration_sections, user_id
+  FROM grading_registration
+  GROUP BY user_numeric_id
+) as sr ON u.user_numeric_id=sr.user_numeric_id
+WHERE u.user_numeric_id=?", array($numeric_id));
+
+            if ($this->course_db->getRowCount() > 0) {
+                $user = $this->course_db->row();
+                if (isset($user['grading_registration_sections'])) {
+                    $user['grading_registration_sections'] = $this->course_db->fromDatabaseToPHPArray($user['grading_registration_sections']);
+                }
+                $details = array_merge($details, $user);
+            }
+        }
```

Comment:
Move this into a function that both `getUserBy*` call with parameter of just the `user_id` please, no reason to duplicate this across both.